resources/wikia/modules/domCalculator.js

```diff
@@ -35,12 +66,81 @@ define('wikia.domCalculator', [
 			topViewport = scrollPosition + globalNavHeight,
 			bottomViewport = scrollPosition + Math.max(doc.documentElement.clientHeight, win.innerHeight || 0);
 
-		return topElement >= (topViewport - elementHeight/2) &&
-			bottomElement <= (bottomViewport + elementHeight/2);
+		return topElement >= (topViewport - elementHeight / 2) &&
+			bottomElement <= (bottomViewport + elementHeight / 2);
+	}
+
+	/**
+	 * Returns element's offset of given element from the left of the page
+	 *
+	 * @param number - number to check
+	 * @param from - range start
+	 * @param length - range length
+	 * @returns {boolean} - true if number is between from and from+length
+	 */
+	function isBetween(number, from, length) {
+		return number >= from && number <= (from + length);
+	}
+
+	/**
+	 * Returns viewport left offset
+	 *
+	 * @returns {number} - offset in pixels
+	 */
+	function getScrollX() {
+		return win.scrollX || win.pageXOffset;
+	}
+
+	/**
+	 * Returns viewport top offset
+	 *
+	 * @returns {number} - offset in pixels
+	 */
+	function getScrollY() {
+		return win.scrollY || win.pageYOffset;
+	}
+
+	/**
+	 * Checks if floating element is in conflict with given element
+	 *
+	 * @param floating - DOM element
+	 * @param element - DOM element
+	 * @returns {boolean} - true if floating is in conflict with element
+	 */
+	function isFloatingInConflict(floating, element) {
+		if (!floating || !element) {
+			return false;
+		}
+
+		floating = {
+			l: getLeftOffset(floating) + getScrollX(),
+			t: getTopOffset(floating) + getScrollY(),
+			w: floating.offsetWidth,
+			h: floating.offsetHeight
+		};
+		element = {
+			l: getLeftOffset(element),
+			t: getTopOffset(element),
+			w: element.offsetWidth,
+			h: element.offsetHeight
+		};
+
+		if (!floating.w || !floating.h || !element.w || !element.h) {
+			return false;
+		}
+
+		return isBetween(floating.l, element.l, element.w) && (
```

Comment:
I don't want to be a pain in the neck but here there are so many calculations and yet there is not a single test. The only way to check is to test it manually.

Reply:
As you wish, I added a bunch of tests :)